slsim/lens.py

```diff
@@ -502,7 +524,48 @@ def point_source_magnitude_molet(self, band, time, **kwargs_molet):
 
         # TODO: in what format should be the 2d source profile be stored (as it is time- and wavelength dependent)
         # TODO: do we create full light curves (and save it in cache) or call it each time
-        return 0
+
+        # we can use this to get intrinsic brightness of a source at image observation
+        # time. Here, we are storing source profile as dictionary of observation time
+        # and magnitude but need to decide what format molet needs. I checked with molet
+        #  and it says format should be in the form given below.
+        source_profile = [
+            {
+                "time": list(self.source.lightcurve_time),
+                "signal": list(
+                    self._point_source_magnitude(
+                        band=band, lensed=False, time=self.source.lightcurve_time
+                    )
+                ),
+            }
+        ]
+        # using source profile, kappa, gamma, kappa_star, image_observed_time molet
+        # should return lightcurve of each images. The lightcurve can be a interpolated
+        # function or a dictionary of observation time and magnitudes in specified band
+        # as given below.
+        # This molet_output is temporary. Once we call molet, this will be actual molet
+        # output.
+        molet_output = [
```

Comment:
can you actually code up to save that as a self variable in cache such that it first used a pre-computed output and only if it does not do that, is going to ask to call molet?

Reply:
Hi @sibirrer, I made it self variable. Is that all I need to do or I need to check whether these self.molet_output contains the lightcurves or not? If it contains use them otherwise call molet?